### ISSUE TITLE
float-compare: false positive not native float64 

### ISSUE THREAD

**Issue by mmorel-35** [Antonboom/testifylint#150]
While applying float-compare on https://github.com/prometheus/prometheus/pull/14334/files#diff-7c1dfea5c568897587621510c04d322f908a8ed40751e592d63d2f8422b6a78dR5926

I had deactivate testifylint here because it suggest to use `InDelta` here but once applied there is an error saying 
```
Parameters must be numerical
```

This is due to SampleValue not being a native float64 
https://github.com/prometheus/common/blob/ab322ea2c291549d457c23476304ed16a4d814a3/model/value_float.go#L32

**Comment by ccoVeille** [comment]
Good catch

**Comment by Antonboom** [comment]
Looks like a bug in `testify`
